### DISCUSSION TITLE
Add the whole language list to the daily leaderboard

### DISCUSSION THREAD

**Discussion by EMRD95** [monkeytypegame/monkeytype#4134] [Ideas]
It would be nice to add the whole language list to the daily leadeboard, but as mentioned by @Miodec in https://github.com/monkeytypegame/monkeytype/issues/4133#issuecomment-1493328935_ "This put a huge load onto the Redis server."

I open this discussion to discuss the feasibility of this project.

By checking the browser network usage, I think the leadeboard retrieves the whole user scores for all modes when loaded, this could be improved by loading only the selected language from the api.
Maybe also adding pagination to retrieve only the first 10 scores.

![image](https://user-images.githubusercontent.com/114953576/229356222-589e2750-3b9b-47c8-8e71-507ebab8db37.png)

**Comment by Miodec** [comment]
Only the currently selected language is requested. This change is not a problem of requesting the data, but about building the leaderboard.

**Reply by EMRD95** [reply]
One leaderboard could be built every 5 minutes? That'd build all leaderboards in less than 24 hours.
Also, instead of 1000 users, only 100 could be used?

**Comment by mikesellt** [comment]
I know this is over a year old now, but could there be a way this could be an option for those running a self-hosted Docker installation?  Maybe not the ENTIRE list, but I'd like to have a leaderboard for the "english_1k," but as of now there isn't a way to get that from what I can tell.  But if it were possible, those running Docker could just edit the leaderboards section of the backend .env file to add it.  It would also be cool to have a third option (having 15, 30, and 60), but I'm sure that would require quite a bit of changes to implement.
